OpenRA.Game/Map/Map.cs

```diff
@@ -285,6 +287,26 @@ public static string ComputeUID(IReadOnlyPackage package)
 			}
 		}
 
+		// PERF This is a way to get MapFormat without expensive yaml parsing
+		public static int GetMapFormat(IReadOnlyPackage p)
+		{
+			var result = "";
+			foreach (var line in p.GetStream("map.yaml").ReadAllLines())
+			{
+				var search = Regex.Match(line, "MapFormat: (.?\\d)");
```

Comment:
Something like `^MapFormat:\\s*(\\d*)\\s*$` would be more robust against false matches.

Reply:
ok